src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.h

```diff
@@ -0,0 +1,197 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace MicrowaveOvenControl {
+
+constexpr uint32_t kDefaultCookTime    = 30;
+constexpr uint8_t kDefaultPowerSetting = 100;
+constexpr uint8_t kDefaultMinPower     = 10;
+constexpr uint8_t kDefaultMaxPower     = 100;
+constexpr uint8_t kDefaultPowerStep    = 10;
+constexpr uint32_t kMaxCookTime        = 65535;
+constexpr uint32_t kMinCookTime        = 1;
+
+class Delegate;
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+public:
+    /**
+     * @brief Creates a Microwave Oven Control cluster instance. The Init() function needs to be called for this instance
+     * to be registered and called by the interaction model at the appropriate times.
+     * @param aDelegate A pointer to the delegate to be used by this server.
+     * Note: the caller must ensure that the delegate lives throughout the instance's lifetime.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the operational state derived cluster to be instantiated.
+     */
+    Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId);
```

Comment:
This relies on someone calling `SetOPInstance` to work correctly, right?  Why is that operational state instance not just a constructor argument?  Or an argument to an Instance Init method?

Reply:
Yes, OpState Instance could be an argument of INIT method of MicrowaveOvenControl, if so, there would be no need to store the OpState instances in global.

previously our thought is to use an OpState array could match with an endpoint id, then get/set OpState instances which could handle if there are muti-endpoints in the device.

Reply:
But we have per-endpoint instances anyway, right?  Why not just store in the instance instead of an out-of-band array?